sdk/identity/azure-identity/CMakeLists.txt

```diff
@@ -127,6 +127,7 @@ az_rtti_setup(
 if(BUILD_TESTING)
   # define a symbol that enables some test hooks in code
   add_compile_definitions(TESTING_BUILD)
+  add_compile_definitions(_azure_TESTING_BUILD)
```

Comment:
I am going to stage the change by adding this "new" macro first, in identity. And then remove/rename the `TESTING_BUILD` across the repo, in one shot.
https://github.com/Azure/azure-sdk-for-cpp/issues/5334